homeassistant/components/zha/core/store.py

```diff
@@ -81,8 +82,8 @@ def async_update_device(self, device: ZHADevice) -> ZhaDeviceEntry:
         ieee_str: str = str(device.ieee)
         old = self.devices[ieee_str]
 
-        if old is not None and device.last_seen is None:
-            return
+        if device.last_seen is None:
+            return old
```

Comment:
Hi @Adminiuga / @dmulcahey

I am not sure of the logic here inside `async_update_device`
`old` cannot ever be `None` so I have removed the check for `if old is not None`

From what I could make out, the code in gateway would be happy to return `None`.
https://github.com/home-assistant/core/blob/aa314a090146b44d32ab2a2754321ae03bf248de/homeassistant/components/zha/core/gateway.py#L553-L556

But the internal code in `async_create_or_update_device` definitely expects a value to be returned.
https://github.com/home-assistant/core/blob/aa314a090146b44d32ab2a2754321ae03bf248de/homeassistant/components/zha/core/store.py#L64-L68